apps/extension/src/background/keyring/keyring.ts

```diff
@@ -592,4 +595,36 @@ export class KeyRing {
     }
     return account.public;
   }
+
+  async generateProof(
+    transferMsg: Uint8Array,
+    generate: (spendingKey: SpendingKey) => Promise<void>
+  ): Promise<void> {
+    await this.vaultService.assertIsUnlocked();
+
+    // We need to get the source address to find the spending key
+    const { sources } = deserialize(Buffer.from(transferMsg), TransferMsgValue);
```

Comment:
Maybe we could use `@heliax/namada-sdk` `deserializeTx` for this, though I think this should work, unless we start passing a batch Tx or multiparty transfer to this